keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/SlackService.kt

```diff
@@ -14,23 +14,24 @@ import org.springframework.stereotype.Component
  */
 @Component
 @EnableConfigurationProperties(SlackConfiguration::class)
-class SlackNotifier(
+class SlackService(
   private val springEnv: Environment,
-  val slackConfig: SlackConfiguration
+  final val slackConfig: SlackConfiguration
 ) {
 
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
 
+  private val configToken = slackConfig.token
+  private val slack = Slack.getInstance()
+
   private val isSlackEnabled: Boolean
     get() = springEnv.getProperty("keel.notifications.slack", Boolean::class.java, true)
 
-
   fun sendSlackNotification(channel: String, blocks: List<LayoutBlock>, token: String? = null) {
     if (isSlackEnabled) {
-      log.debug("starting slack notification")
-      val slack = Slack.getInstance()
+      log.debug("sending slack notification for channel $channel")
```

Comment:
what do you mean by ` increment a counter metric`?

Reply:
Here's an example: https://github.com/spinnaker/keel/blob/master/keel-core/src/main/kotlin/com/netflix/spinnaker/keel/telemetry/TelemetryListener.kt#L34-L45

Lorin means publishing a metric for this so that we can graph it in atlas / alert on things like a huge increase in # of messages sent

Reply:
ok i've added some counter metrics, hope it looks ok. will add more when I'll add more logic!